contributor-docs/adrs/adr-003-prop-norms.md

```diff
@@ -0,0 +1,53 @@
+# ADR 003: Prop norms in Primer React Components
+
+## Status
+
+In progress / draft / pending?
+
+## Context
+
+- PRC APIs matter
+- Consistency is important
+- There’s a gap between where we want to be and where we
+- This document aims to codify our best practices for component APIs and unify disparate conversations
+  - In the course of some refactorings, we may have intermediate states where there will added be value in a clear documentation of which parts of our API reflect the best practices.
+
+## Decision
+
+### 🟢 `sx`
+
+All components that ultimately render to the DOM should accept an `sx` prop. The `sx` prop (of type `SystemStyleObject`) should set styles for the root HTML element rendered by the component. It may make sense to have an additional prop to style a descendant element (e.g., `innerSx`).
+
+### 🟢 `ref`
+
+All components that ultimately render to the DOM should accept a `ref` prop. That `ref` prop should be passed to the root HTMLElement rendered by the component.
+
+### 🟡 `as`: Limited
+
+Only components with a clear need for polymorphism should accept an `as` prop. Reasonable cases include:
+
+- Components that need functionality from the `as` prop, like a button
+- Components whose accessibility are improved by using semantically appropriate HTML elements
+
+### 🟡 DOM props: Limited
+
+All components that accept an `as` prop should accept the props for the element specified by the as prop. _Additionally_, some other elements (e.g., `<TextInput>`) should accept the props for their root HTML element when those props are fundamental to the component’s function.
+
+### 🔴 System props
+
+Components should not accept styled system props (except our utility components: `Box` and `Type`)
```

Comment:
I wonder if we can explain why utility components get system props and the rest of the components don't. 

This is worded badly but something like... The primary function of utility components is to provide a convenient API for writing custom styles using design system values so it doesn't make sense to bury that API in the `sx` prop. Non-utility components' purpose is to implement specific design patterns. Custom styling is not their primary function so we don't want them to have a first-class API (i.e. system props) for overriding styles.

Reply:
How do y'all feel about using sx prop in the case of utility components as well for the sake of the "one true way" to use tokens from the system?

Reply:
I'm open to that!

Reply:
Ooh, that's interesting. A couple questions that might inform whether `sx` is sufficient for our utility components:
- Would the slight loss of convenience lead to users taking shortcuts that they wouldn't otherwise take? Would a user decide to apply styles directly or reach down into styled-components to avoid the clunkiness of going through `sx`?
- Would a potential user decide not to use PRC because of the loss of convenience? Would they decide to use PRC because of the more consistent API?